core/kazoo_number_manager/src/knm_maintenance.erl

```diff
@@ -89,14 +91,61 @@ fix_account_numbers(Account = ?NE_BINARY) ->
 -spec migrate() -> 'ok'.
 migrate() ->
     AccountDbs = kapps_util:get_all_accounts(),
-    foreach_pause_in_between(?TIME_BETWEEN_ACCOUNTS_MS, fun migrate/1, AccountDbs).
+    foreach_pause_in_between(?TIME_BETWEEN_ACCOUNTS_MS, fun migrate/1, AccountDbs),
+    erase(callflow_DIDs),
+    erase(trunkstore_DIDs),
+    migrate_unassigned_numbers().
 
 -spec migrate(ne_binary()) -> 'ok'.
-migrate(AccountDb) ->
+migrate(Account) ->
+    AccountDb = kz_util:format_account_db(Account),
     fix_account_numbers(AccountDb),
     _ = kz_datamgr:del_doc(AccountDb, <<"phone_numbers">>),
     'ok'.
 
+-spec migrate_unassigned_numbers() -> 'ok'.
+-spec migrate_unassigned_numbers(ne_binary(), integer()) -> 'ok'.
+migrate_unassigned_numbers() ->
+    ?LOG("********** fixing unassigned numbers **********", []),
+    NumberDbs = knm_util:get_all_number_dbs(),
+    foreach_pause_in_between(?TIME_BETWEEN_ACCOUNTS_MS
+                            ,fun migrate_unassigned_numbers/1
+                            ,NumberDbs
+                            ),
+    ?LOG("********** finished fixing unassigned numbers **********", []).
+
+migrate_unassigned_numbers(<<?KNM_DB_PREFIX_ENCODED, _/binary>> = NumberDb) ->
+    ?LOG("########## start fixing ~s ##########", [NumberDb]),
+    migrate_unassigned_numbers(NumberDb, 0),
+    _ = knm_phone_number:push_stored(), %% Bulk doc writes
+    ?LOG("########## done fixing ~s ##########", [NumberDb]);
+migrate_unassigned_numbers(<<?KNM_DB_PREFIX_encoded, Suffix/binary>>) ->
+    migrate_unassigned_numbers(<<?KNM_DB_PREFIX_ENCODED, Suffix/binary>>);
+migrate_unassigned_numbers(<<?KNM_DB_PREFIX, Suffix/binary>>) ->
+    migrate_unassigned_numbers(<<?KNM_DB_PREFIX_ENCODED, Suffix/binary>>);
+migrate_unassigned_numbers(Number) ->
+    migrate_unassigned_numbers(knm_converters:to_db(Number)).
+
+migrate_unassigned_numbers(NumberDb, Offset) ->
+    ViewOptions = [{limit, kz_datamgr:max_bulk_insert()}
+                  ,{skip, Offset}
```

Comment:
nope, it is setting in the case clause to `Offset + Length`. For the first iteration, it's okay as well and will start at beginning of view result.